azurerm/resource_arm_dev_test_windows_virtual_machine.go

```diff
@@ -0,0 +1,295 @@
+				// The name must be between 1 and 15 characters, cannot be entirely numeric, and cannot contain most special characters
+			},
+
+			"lab_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DevTestLabName(),
+			},
+
+			// There's a bug in the Azure API where this is returned in lower-case
+			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/3964
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
+
+			"location": locationSchema(),
+
+			"size": {
+				Type:     schema.TypeString,
+				Required: true,
+				// since this isn't returned from the API
+				ForceNew: true,
+			},
+
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"password": {
+				Type:     schema.TypeString,
+				Required: true,
+				// since this isn't returned from the API
+				ForceNew: true,
+			},
+
+			"storage_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Standard",
+					"Premium",
+				}, false),
+			},
+
+			"lab_subnet_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				// since this isn't returned from the API
+				ForceNew: true,
+			},
+
+			"lab_virtual_network_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				// since this isn't returned from the API
+				ForceNew: true,
+			},
+
+			"allow_claim": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"disallow_public_ip_address": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"gallery_image_reference": azure.SchemaDevTestVirtualMachineGalleryImageReference(),
+
+			"inbound_nat_rule": azure.SchemaDevTestVirtualMachineInboundNatRule(),
+
+			"notes": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"tags": tagsSchema(),
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"unique_identifier": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmDevTestWindowsVirtualMachineCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).devTestVirtualMachinesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for DevTest Windows Virtual Machine creation")
+
+	name := d.Get("name").(string)
+	labName := d.Get("lab_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	allowClaim := d.Get("allow_claim").(bool)
+	disallowPublicIPAddress := d.Get("disallow_public_ip_address").(bool)
+	labSubnetName := d.Get("lab_subnet_name").(string)
+	labVirtualNetworkId := d.Get("lab_virtual_network_id").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	notes := d.Get("notes").(string)
+	password := d.Get("password").(string)
```

Comment:
Same as above with doing a d.GetOk instead

Reply:
unfortunately both password and ssh key can be specified (like in a VM) afaik